substrate/frame/support/src/traits/tokens/fungible/union_of.rs

```diff
@@ -925,3 +925,28 @@ impl<
 		}
 	}
 }
+
+impl<
+		Left: fungible::Inspect<AccountId>,
+		Right: fungibles::Inspect<AccountId> + fungibles::Refund<AccountId>,
```

Comment:
Why is this adapter requiring heterogeneous types for Left and Right?  
Why is the left side just assumed to not be Refund?

Reply:
yes, we ether assume there is no refund for `fungible`, or we introduce a no-op impl for `fungible`, which is balance pallet,  with asset id `()`, or we have to introduce one more refund trait without asset id. Given this, I made a chose for no-op in adapter (no refund trait for fungible, no adapter for it). I am open to go for another solution, as I do not see an obvious winner here.